### PR TITLE
fix(autoware_freespace_planning_algorithms): fix shadowVariable

### PR DESCRIPTION
**Pull request by kobayu858**
## Description
This is a fix based on cppcheck shadowVariable warnings

```
planning/autoware_freespace_planning_algorithms/src/astar_search.cpp:308:37: style: Local variable 'pose' shadows outer variable [shadowVariable]
    geometry_msgs::msg::PoseStamped pose;
                                    ^

planning/autoware_freespace_planning_algorithms/src/astar_search.cpp:313:21: style: Local variable 'pw' shadows outer variable [shadowVariable]
    PlannerWaypoint pw;
                    ^

planning/autoware_freespace_planning_algorithms/src/rrtstar_core.cpp:186:19: style: Local variable 'node_best_parent' shadows outer variable [shadowVariable]
    NodeSharedPtr node_best_parent;
                  ^
```

## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/autoware_freespace_planning_algorithms/src/astar_search.cpp [hunk 1]
```diff
@@ -305,15 +305,15 @@ void AstarSearch::setPath(const AstarNode & goal_node)
...
 
     // PlannerWaypoint
-    PlannerWaypoint pw;
-    pw.pose = pose;
-    pw.is_back = node->is_back;
-    waypoints_.waypoints.push_back(pw);
+    PlannerWaypoint node_pw;
+    node_pw.pose = node_pose;
+    node_pw.is_back = node->is_back;
+    waypoints_.waypoints.push_back(node_pw);
```
Comment (veqcc) [inline planning/autoware_freespace_planning_algorithms/src/astar_search.cpp:316]: @kobayu858 
How about embracing `// push exact goal pose first` code block with `{}` ?

Comment (kobayu858) [reply planning/autoware_freespace_planning_algorithms/src/astar_search.cpp]: Thanks for the advice. I made the change as it looks like I can resolve the warning that way.

**Review by veqcc — APPROVED** [review]

**Comment by kobayu858** [comment]
@kosuke55 @TakaHoribe @takayuki5168 

Please confirm the PR. Thank you very much.

**Review by kosuke55 — APPROVED** [review]

**Comment by kobayu858** [comment]
Thanks for the merge.

**Pull request merged**